navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -158,6 +159,54 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
         legJson.put("distance", Helper.round(distance, 1));
     }
 
+
+    /**
+     * fix the first IntersectionDetail.
+     *
+     * The first Intersection of the first step should only have 1 bearing and one "out" entry
+     *
+     */
+    private static List<PathDetail> fixFirstIntersectionDetail( List<PathDetail> intersectionDetails) {
+        List<PathDetail> list = new ArrayList<>();
+
+        // job1: find out the interesting part of the intersectionDetails
+        if (intersectionDetails.size() < 2) {
+            // LOG Error? User didnt ask for intersections ?
+            return intersectionDetails;
+        }
+
+        Map<String, Object> newFirstIntersection = new HashMap<>();
+
+        final Map<String, Object> firstItersectionMap = (Map<String, Object>) intersectionDetails.get(0).getValue();
+
+        int out = (int) firstItersectionMap.get("out");
+        newFirstIntersection.put( "out", 0); // we will have only one entry
+
+        // bearings
+        List<Integer> firstBearingsList = (List<Integer>) firstItersectionMap.get("bearings");
+        List<Integer> firstBearings = new ArrayList<>();
+        firstBearings.add(firstBearingsList.get(out));
+        newFirstIntersection.put( "bearings", firstBearings);
+
+        // entries
+        final List<Boolean> firstEntriesList = (List<Boolean>) firstItersectionMap.get("entries");
+        List<Boolean> firstEntries = new ArrayList<>();
+        firstEntries.add( firstEntriesList.get(out));
+        newFirstIntersection.put( "entries", firstEntries);
+
+        // new first Pathdetail
+        PathDetail firstPathDetail =  new PathDetail( newFirstIntersection);
+        firstPathDetail.setFirst( intersectionDetails.get(0).getFirst());
+        firstPathDetail.setLast( intersectionDetails.get(0).getLast());
+        list.add( firstPathDetail);
+
+        // copy over all elements but the first and last
+        for (int i = 1; i < intersectionDetails.size() - 1; i++) {
+            list.add(intersectionDetails.get(i) );
```

Comment:
I am not sure if we need to copy the list?

Why is the last intersection removed? Especially if the method is called `fixFirstIntersectionDetail`?

Reply:
OOOOPS!

Reply:
The copy of intersectionDetails is done because it is in the end only a reference to the ResponsePath from the GH routing. If we modify it I am not sure if some dependent algo my fail over some assumptions that it is in the GH way and not Mapbox Way.

Reply:
But maybe I am overly cautios

Reply:
I took this approach not to clone, makes the code easier.